packages/rum-core/src/domain/plugins.ts

```diff
@@ -0,0 +1,27 @@
+import type { RumPublicApi } from '../boot/rumPublicApi'
+import type { RumInitConfiguration } from './configuration'
+
+export interface RumPlugin {
+  name: string
+  serializeConfiguration?(): Record<string, unknown>
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏What is the serializeConfiguration meant for? Is it only to be passed in the telemetry config?

Reply:
Yes, but I agree the name isn't great, see https://github.com/DataDog/browser-sdk/pull/2809#discussion_r1639569044